Improve UX for cluster scoped private registry config

catherineluse opened the pull request:
Addresses these issues:

- https://github.com/rancher/dashboard/issues/6919
- https://github.com/rancher/dashboard/issues/5146
- https://github.com/rancher/dashboard/issues/6875

Previously, we showed private registry options with three radio buttons in the cluster edit form for RKE2/K3s. We got reports that this form was confusing to users:
<img width="941" alt="Screenshot 2022-11-19 at 5 44 57 PM" src="https://user-images.githubusercontent.com/20599230/202877461-1d343ce4-edbc-4864-9b12-4f415b928214.png">

To improve clarity, this PR replaces the radio buttons with a checkbox and explains what the default option is above the checkbox.

If the checkbox is selected, the basic registry config options are displayed. If "Show Advanced" is clicked, the mirroring and advanced auth options are displayed. I have also added more UI text and docs links to help tamp down on the confusion.


Default view: 

<img width="1146" alt="Screenshot 2022-11-19 at 5 33 45 PM" src="https://user-images.githubusercontent.com/20599230/202877063-f7793dae-cf94-4cf6-b265-f53d06a94b76.png">

If cluster registry is enabled:
<img width="1144" alt="Screenshot 2022-11-19 at 5 34 41 PM" src="https://user-images.githubusercontent.com/20599230/202877076-95501447-1aca-48f7-afc9-a95214964a25.png">

If advanced options are expanded (docs link changes to K3s docs if a K3s version is selected):

<img width="874" alt="Screenshot 2022-11-19 at 5 42 39 PM" src="https://user-images.githubusercontent.com/20599230/202877398-9f2677da-b712-4f64-855f-50bd86aa4d1d.png">

catherineluse commented on shell/edit/provisioning.cattle.io.cluster/rke2.vue:
```diff
@@ -267,30 +268,31 @@ export default {
+      systemRegistry:                  null,
+      registryHost:                    null,
+      showCustomRegistryInput:         false,
+      showCustomRegistryAdvancedInput: false,
```
the new values are:
```
showCustomRegistryInput:         false,
      showCustomRegistryAdvancedInput: false,
```

mantis-toboggan-md reviewed (changes requested):
I have a couple suggestions for the SelectOrCreateAuthSecret component, but I know that is pre-existing code so if you want me to file a separate issue let me know.
I think it'd be nice if the style aligned more with the namespace dropdown:
![Screen Shot 2022-11-22 at 12 40 05 PM](https://user-images.githubusercontent.com/42977925/203406667-d374c0e5-6d76-4565-af50-3f2a5ad463e4.png)
![Screen Shot 2022-11-22 at 12 41 47 PM](https://user-images.githubusercontent.com/42977925/203406671-4863939b-0f22-49a9-b4e2-7d5f44aa7413.png)

My other issue with the component is the way it loads data (all secrets, sometimes all norman cloud creds too). It's using the `Loading` component, which blocks the user from interacting with the entire page until the data needed for a single dropdown is ready. The experience is a little nicer if the LabeledSelect `loading` prop is used instead (ie add`:loading="$fetchState.pending"`)

https://user-images.githubusercontent.com/42977925/203412965-be4e73d6-d5b9-4bb8-a99e-026828832dad.mov

There is a minor style issue with the 'x' button for configs:
![Screen Shot 2022-11-22 at 12 43 48 PM](https://user-images.githubusercontent.com/42977925/203407068-d60d7389-1929-4bd0-b515-6517efb8044f.png)

I also noticed that when I create a cluster with a registry auth config defined, then edit the cluster to remove that config, the change isn't saved.

catherineluse commented:
I've made these changes:

- Changed styling of the menu to create or select secret to be consistent with the dropdown for creating or selecting a namespace.
- Used loading prop instead of loading component.
- Fixed X button size/overlap issue.
- Removed "configs section" - it now says "Define the TLS and credential configuration for each registry hostname and mirror." This language is updated to convey that it includes the simple basic auth config in the list - you should ALL registry credential secrets and TLS options there, including the basic one. (Basically, clicking advanced options is revealing everything as it is in YAML.)
- When you delete an auth config, the change is now saved.

mantis-toboggan-md reviewed (approved):
LGTM

mantis-toboggan-md commented:
I just noticed that the style change to ArrayListGrouped looks off elsewhere in the UI:
![Screen Shot 2022-12-07 at 1 57 09 PM](https://user-images.githubusercontent.com/42977925/206294248-3fd6ac79-6100-4117-858a-04c8e83fb4d2.png)

mantis-toboggan-md commented on shell/components/form/ArrayListGrouped.vue:
```diff
@@ -106,10 +113,11 @@ export default {
 
         top: 0;
         right: 0;
+        margin-right: -2.25em;
```
I think since this component is used full-width elsewhere it might be better to do something like add more padding to `.info-box`

mantis-toboggan-md commented on shell/edit/provisioning.cattle.io.cluster/rke2.vue:
```diff
@@ -2196,48 +2217,82 @@ export default {
+              <AdvancedSection
+                :is-open-by-default="showCustomRegistryAdvancedInput"
+                :mode="mode"
+                class="advanced"
```
Sorry I didn't notice this on my first review, but the contents of this tab don't fill the full width of the section like the other tabs' inputs do. Adding `col span-12` here and changing line 2224 to `span-12` would make this tab's content full-width.
![Screen Shot 2022-12-07 at 2 16 48 PM](https://user-images.githubusercontent.com/42977925/206298341-3d62f9fc-3f5f-4e8a-9d3e-3e34ff1a23f3.png)

catherineluse commented:
I've changed the columns to full width and adjusted the padding for the X button:

![Screenshot 2022-12-08 at 2 55 00 PM](https://user-images.githubusercontent.com/20599230/206575806-7549013b-8698-4da8-997e-846e2acf827c.png)

Here's the padding in the Ingress rules context:

![Screenshot 2022-12-08 at 2 51 20 PM](https://user-images.githubusercontent.com/20599230/206575878-d1743eb1-e9d0-4e41-a6cb-f9d74de22176.png)
